devito/propagator.py

```diff
@@ -200,7 +201,7 @@ def sympy_to_cgen(self, stencils):
                 # TODO: add support for double precision
                 self.add_local_var(name, np.float32)
                 # TODO: undo precision enforcing
-                factors.append(cgen.Assign(name, str(ccode(expr.xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
+                factors.append(cgen.Assign(name, str(ccode(self.time_substitutions(expr).xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
```

Comment:
`convert_equality_to_cgen` does not care about `save` and always call `time_substitutions` on the stencil. Why would it be different for the factors?